### PR TITLE
Server-side paywall optimization: refactor client and server-side flows

### PR DESCRIPTION
**Pull request by teiswit**
- Add missing Regwall scenario for anonymous user in server-side flow
- Improve modularity for server vs. client paywall flows
- Update validation rule for server-side flow (only userState is required)

Internal tracking: b/263350663

### CODE REVIEW

File: src/runtime/extended-access/gaa-metering.ts [hunk 1]
```diff
@@ -454,16 +474,26 @@ export class GaaMetering {
...
+    // For server-side paywall, userState needs to be provided.
+    if ('userState' in params && typeof params.userState !== 'object') {
+      debugLog(`userState is not an object`);
+      noIssues = false;
+    } else if (
+      params.paywallType == PaywallType.SERVER_SIDE ||
+      params.showcaseEntitlement
+    ) {
+      if (!('userState' in params)) {
+        debugLog('userState needs to be provided');
```
Comment (ChrisAntaki) [inline src/runtime/extended-access/gaa-metering.ts:488]: Could you add a test covering this?

Comment (teiswit) [reply src/runtime/extended-access/gaa-metering.ts:488]: Done - added test cases to cover this scenario.

**Comment by teiswit** [comment]
@ChrisAntaki PTAL

**Review by ChrisAntaki — APPROVED** [review]

**Comment by ChrisAntaki** [comment]
Looks good!

**Pull request merged**